matchmaking.py

```diff
@@ -0,0 +1,61 @@
+import random
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Matchmaking:
+    def __init__(self, li, config):
+        self.li = li
+        self.variants = config["challenge"]["variants"].copy()
+        if "fromPosition" in self.variants:
+            self.variants.pop(self.variants.index("fromPosition"))
+        self.matchmaking_cfg = config["matchmaking"]

```

Comment:
This line needs to be `config.get("matchmaking")` in case the user does not have that section in their config.yml.